_drafts/javascript.md

```diff
@@ -0,0 +1,294 @@
+
+### Add script for JavaScript linting
+
+The only tool I would like to introduce as part of this proposal is a linting tool which will be run as part of `npm test`.
+
+I suggest [standard](https://standardjs.com/). We’ve been using it in all our projects at Hoodie and Neighbourhoodie since 2+ years and never looked back. It's a zero-configuration JavaScript linter.
+
+### Update documentation
+
+Remove any reference of CoffeeScript from the documentation.
+
+## Backward compatibility
+
+Only breaking change is dropping support of Node 0.10 and 0.12.
+
+## Alternatives considered
+
+- **[CoffeeScript 6.0](https://github.com/coffeescript6/discuss)**
+
+  As stated in the [Background section](https://github.com/coffeescript6/discuss#background), JavaScript caught up with most of the features that only CoffeeScript offered in the past.
+
+  The argument for the clear syntax is valid. But we can enforce that with a JavaScript linter.
+
+  The main reason we decided to convert Hubot to JavaScript is to make it easier to contribute to for a wider range of people.
+
+- **Continue to support Node <v4**
+
+  Node versions smaller than v4 are no longer maintained and have known security vulnerabilities. The only major operating system which distributes a no longer maintained Node version by default is Debian, and there are alternative ways to install a recent Node version, see [Debian and Ubuntu based Linux distributions](https://nodejs.org/en/download/package-manager/#debian-and-ubuntu-based-linux-distributions).
+
+  Mikeal Rogers (Node Foundation) says about Debian’s official Node packages:
+
+  > The "official" repositories are outdated and even insecure. We need to treat them as legacy and move on.
+
+  The Node Foundation has tried to get Debian to adapt modern Node versions like "[pretty much every other repository for packages takes modern versions.](https://twitter.com/mikeal/status/869646796888330240)" but could not succeed.
+
+  See more responses to my question regarding old Node versions at https://twitter.com/gr2m/status/869305267464306689
+
+- **Dropping support for Node <v7.6**
+
+  _**Note:** this is still up for discussion even though the text below is written in past tense. But if we decide against, I would like to document why. Same if we decide in favour of requiring Node v7.6+_
+
+  Node 7.6 introduced native support for async/await. For asynchronous code like much of Hubot is, this is a very nice feature. It makes the code more readable and hence more accessible for contributors which is a main objective of the Hubot community.
+
+  The reason we decided against it is that the complexity of Hubot core is not too high, the code can be made very readable even when using Promises. Once the one-time conversion from CoffeeScript to JavaScript is done, only smaller parts of the code base will be touched, the effect of async/await vs. promises will be limited.
```

Comment:
Yes, Hubot user can take full advantage for the Node version they run their own Hubot instance in. If Hubot core adds Promise-based APIs then users which run their Hubot in Node 7.6+ can use async/await as these are sugar APIs to work with Promises. At the same time the core would continue to work with Node v4, so we don’t break Hubot instances that require this older Node version. Does that answer your question?